packages/node_modules/@ciscospark/widget-message/src/selector.js

```diff
@@ -27,33 +24,22 @@ export const getMessageWidgetProps = createSelector(
   [
     getSparkInstance,
     getSparkState,
-    getAvatars,
-    getActivity,
     getConversation,
-    getFlags,
-    getMercury,
-    getUsers,
-    getWidget,
-    getSpace,
-    getFeatures
+    getParticipants,
+    getActivities,
+    getSpace
   ],
-  (sparkInstance, sparkState, ...props) => {
+  // eslint-disable-next-line max-params
+  (sparkInstance, sparkState, conversation, participants, activities, space) => {
     const propsObject = {};
-    const propNames = [
-      `avatar`,
-      `activity`,
-      `conversation`,
-      `flags`,
-      `mercury`,
-      `user`,
-      `widgetMessage`,
-      `space`,
-      `features`
-    ];
-    props.map((p, i) => {
-      propsObject[propNames[i]] = p;
-      return p;
-    });
+
+    propsObject.activities = activities.toJS();
+    propsObject.activityCount = activities.count();
+    propsObject.lastActivity = activities.last();
+    propsObject.firstActivity = activities.first();
+    propsObject.conversationId = conversation.get(`id`);
+    propsObject.participants = participants.toJS();
+    propsObject.space = space;
```

Comment:
would this be better as a single constant set? e.g.
``` js
const propsObject = {
  activities: activities.toJS(),
  activityCount: activities.count(),
  ...
};
```

Reply:
Yea, that looks cleaner, not sure why I went this route except maybe debugging. Will change it.